sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/resource/Ec2Resource.java

```diff
@@ -212,9 +212,9 @@ private static void readTo(@Nullable InputStream is, ByteArrayOutputStream os)
     if (is == null) {
       return;
     }
-    int b;
-    while ((b = is.read()) != -1) {
-      os.write(b);
+    byte[] buf = new byte[8192];
+    while (is.read(buf) != -1) {
+      os.write(buf);
```

Comment:
You can't write the full buffer here. You need to check how many bytes you actually read into the buffer, which may be less.